xenvm/lvresize.ml

```diff
@@ -51,7 +51,11 @@ let resize_remotely info vg_name lv_name size =
 let resize_locally allocator vg_name lv_name size =
   let dm_name = Mapper.name_of vg_name lv_name in
   let s = Lwt_unix.socket Lwt_unix.PF_UNIX Lwt_unix.SOCK_STREAM 0 in
-  Lwt_unix.connect s (Unix.ADDR_UNIX allocator)
+  (try
+     Lwt_unix.connect s (Unix.ADDR_UNIX allocator)
+   with e ->
+     Lwt_unix.close s;
+     raise e)
```

Comment:
I think you need to use `Lwt.fail` for exceptions in lwt blocks

Reply:
Exactly.